app/controllers/channels_controller.rb

```diff
@@ -8,6 +8,9 @@ def show
     @members = workspace.users
     @channel_members = channel.users
     @messages = channel.messages.order(created_at: :asc)
+  rescue ActiveRecord::RecordNotFound => _e
+    general_channel = workspace.channels.find_by(name: 'general')
+    redirect_to workspace_channel_path(workspace, general_channel)
```

Comment:
will there always be a `general` channel on all workspaces? What if an admin/owner wants to remove it?

Reply:
The 'general' channel is the basic channel of any workspace, it's public and it can't be deleted for any user of any role. There is a policy that refuses to delete a 'general' channel, no matter the role of the user.